server/src/resolvers/Mutation/pyrog.ts

```diff
@@ -227,5 +228,41 @@ export const pyrogMutation = {
     //     throw new PermissionError()
     // }
     return context.client.deleteAttribute({ id: id });
+  },
+  async upsertCredential(
+    parent,
+    { login, password, host, port, type, database, sourceId },
+    context: Context
+  ) {
+    getUserId(context);
+    const encryptedPassword = await bcrypt.hash(password, 10);
+    const credential = await context.client
+      .source({ id: sourceId })
+      .credential();
+    console.log(credential);
```

Comment:
nit: remove console.log

Reply:
Oops, my bad. Useless `console.log` are removed in 1460525